src/Domain/LeanCode.DomainModels/Ids/TypedIdConverter.cs

```diff
@@ -45,4 +78,16 @@ public class GuidTypedIdConverter<TId> : JsonConverter<TId>
 
     public override void Write(Utf8JsonWriter writer, TId value, JsonSerializerOptions options) =>
         writer.WriteStringValue(value.Value);
+
+    public override TId ReadAsPropertyName(
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options
+    ) =>
+        reader.GetString() is string s && Guid.TryParse(s, out var id)
```

Comment:
Why not [GetGuid](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.getguid?view=net-7.0)?